Make masked_scatter core aten

jackzhxng opened the pull request:
Summary: Making `masked_scatter` core aten since it is hard to decompose and we now have a portable kernel for it

Test Plan: N/A

Differential Revision: D64368725

larryliu0820 commented on aten/src/ATen/native/native_functions.yaml:
```diff
@@ -8010,6 +8010,7 @@
     CUDA: masked_scatter__cuda
     MPS: masked_scatter__mps
   autogen: masked_scatter.out
+  tags: core
```
Does it work if we mark the functional variant `masked_scatter` as core?

Commit:
Make masked_scatter core aten (#137949)

Summary:
Making `masked_scatter` core aten since it is hard to decompose and we now have a portable kernel for it


Test Plan: N/A

Differential Revision: D64368725

Pulled By: dvorjackz

dvorjackz commented:
@pytorchbot merge
